### PR TITLE
COMPRESS-566 allow gzip buffer size to be configured

### PR DESCRIPTION
**Pull request by bokken**

### CODE REVIEW

File: src/main/java/org/apache/commons/compress/compressors/gzip/GzipParameters.java [hunk 1]
```diff
@@ -118,4 +120,22 @@ public int getOperatingSystem() {
     public void setOperatingSystem(final int operatingSystem) {
         this.operatingSystem = operatingSystem;
     }
+
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/compress/compressors/gzip/GzipParameters.java:123]: You're missing a Javadoc comment here.

**Review by garydgregory — CHANGES_REQUESTED** [review]
See comment.

File: src/main/java/org/apache/commons/compress/compressors/gzip/GzipParameters.java (hunk 1 above)
Comment (bokken) [reply src/main/java/org/apache/commons/compress/compressors/gzip/GzipParameters.java:123]: The rest of the attributes in the class only had javadoc for setter method. I had started writing javadoc but removed it to match the rest. I can certainly add.

File: src/main/java/org/apache/commons/compress/compressors/gzip/GzipParameters.java [hunk 2]
```diff
@@ -118,4 +120,28 @@ public int getOperatingSystem() {
...
+    /**
+     * Sets size of the buffer used to retrieve compressed data from
+     * {@link Deflater} and write to underlying {@link OutputStream}.
+     * 
+     * @param bufferSize the bufferSize to set. Must be a positive value.
+     * @since 1.21
+     */
+    public void setBufferSize(int bufferSize) {
+        if (bufferSize <= 0) {
```
Comment (PeterAlfredLee) [inline src/main/java/org/apache/commons/compress/compressors/gzip/GzipParameters.java:142]: Do we need a high limit check for `bufferSize` here?

Comment (bokken) [reply src/main/java/org/apache/commons/compress/compressors/gzip/GzipParameters.java:142]: I personally do not see the value, but can add if desired. 
There is certainly some threshold where larger values are counter productive, but my opinion is consumers can determine that on their own.

**Comment by bokken** [comment]
Do I need to do anything else here @garydgregory ?

**Comment by bodewig** [comment]
@garydgregory @PeterAlfredLee is there anythin gleft that needs to be done before merging this?

**Comment by PeterAlfredLee** [comment]
This looks good to me. :-)

**Pull request merged**